Fixes for building the 2.3 release

kmod opened the pull request:
:tada:

kmod committed:
2.2 -> 2.3

kmod committed:
install perf into the docker containers

kmod committed:
Add an opt-version target of the test suite

kmod committed:
Add a test that installs and rebuilds all packages

undingen committed:
release: allow perf to run inside the container and hack around perf version The linux-tools package of the newer host kernel is not available, but it seems like using the older linux-tools perf works with the new kernel. The new 'perf' symlink just overwrites the version check.

kmod committed:
Also install perf for the current kernel

in case your current kernel is out of date with the repo latest

kmod committed:
Makefile: avoid construct not supported on 16.04

kmod committed:
do parallel make in make_release.sh

kmod committed:
Add arg to build all release packages in parallel

kmod committed:
let the pgo task fail again

test_os often fails for me when run inside the debian build system

ubuntu lets the pgo task fail, but I had turned that off because I kept breaking the pgo task and we ended up with pgo-less builds. that hasn't happened in a while so I prefer to let it fail again.

kmod committed:
Change from using perf2bolt to bolt -p

https://github.com/facebookincubator/BOLT/issues/187

undingen committed:
release: update library dependencies
install the libraries from the README in the docker container.

kmod committed:
make_release.sh: allow specifying which release to build

kmod committed:
I think we don't need to do this if we dont send the build/ directory

kmod committed:
Packaging fix: make this a relative symlink rather than a absolute

Otherwise the installed link is a link to the location it was built in the docker container

kmod committed:
workaround: looks like this bolt flag causes issues but only on old ubuntus

kmod committed:
bench_release.sh: install nginx and workaround pyperformance==1.0.2 issue

pyperformance 1.0.2 does not use the venv it creates when a different interpreter is specified using -p.

kmod committed:
fix macrobenchmark runner typo

kmod committed:
Unpatch pip

we had patched this to write out a shebang of pyston3.8, but with the new name change
we want the original behavior again

kmod committed:
Change our install scheme

previously we installed as pyston3.8, symlinked to pyston and pyston3
now we install to python3.8-pyston2.2, symlinked to pyston and pyston3

and instead of pip-pyston3.8, install to pip-pyston2.3

kmod committed:
run_profile_task.py: try the benchmark a couple times

it shouldn't be a big problem if we end up running a slightly longer profile task,
but it's super annoying if a 2-hour package build fails because the profile task
had a transient issue

gurnec commented:
Forgive me if I'm wrong, but it looks like the [recently released .deb packages](https://github.com/pyston/pyston/releases/tag/v2.3) are based on this branch, even though this branch has yet to be merged into pyston_master?

Likewise, it looks like the [v2.3 tag in this repo](https://github.com/pyston/pyston/tree/v2.3) does not reference the source used to build those .debs?

kmod commented:
Sorry yes:
- the .deb packages are based on the code in this pr
- github automatically created the v2.3 tag and put it on the master branch (it now points to this pr)

Did this cause an issue? I thought it would be fine to put out the release before merging the PR but if there's an issue with that I apologize.

gurnec commented:
> Did this cause an issue? I thought it would be fine to put out the release before merging the PR but if there's an issue with that I apologize.

No issue as far as I'm concerned, I was just a bit confused, and I'm sorry if I implied otherwise.

I was looking for a changelog in this repo, and noticed that the version number between the source and binaries differed. Thanks for clarifying though, and sorry for bugging you on the weekend!

kmod committed:
create a symlink to the proper portable release file

untested

I uploaded the 20.04 portable dir instead of the 16.04 one,
and this should hopefully avoid that mistake in the future
